### PR TITLE
add interpreter to handle pwsh and powershell to powershell_script

### LINKED ISSUE #10428: support pwsh interpreter in powershell_script resource
Related to epic #10330 

By default, `powershell_script` will continue to use powershell.exe, but users can optionally specify `pwsh` as an interpreter. This will require pwsh (cross platform powershell) to be installed and on the path.

### PR DESCRIPTION
**Pull request by mwrock**
closes #10428

Signed-off-by: mwrock <matt@mattwrock.com>

### CODE REVIEW

File: lib/chef/resource/powershell_script.rb [hunk 1]
```diff
@@ -22,11 +22,17 @@ class Resource
...
+      set_guard_inherited_attributes(:interpreter)
+
       provides :powershell_script, os: "windows"
 
       property :flags, String,
         description: "A string that is passed to the Windows PowerShell command"
 
+      property :interpreter, [:powershell, :pwsh],
+        default: :powershell,
+        description: "the interpreter type, `:powershell` or `:pwsh` (PowerShell Core)"
```
Comment (tas50) [inline lib/chef/resource/powershell_script.rb:34]:
```suggestion
        description: "The interpreter type, `:powershell` or `:pwsh` (PowerShell Core)"
```

File: lib/chef/provider/powershell_script.rb [hunk 2]
```diff
@@ -54,7 +54,14 @@ def command
...
         # Powershell.exe is always in "v1.0" folder (for backwards compatibility)
-        Chef::Util::PathHelper.join(basepath, "WindowsPowerShell", "v1.0", interpreter)
+        # pwsh is the other interpreter and we will assume that it is on the path.
+        # It will exist in different folders depending on the installed version.
+        # There can also be multiple versions installed. Depending on how it was installed,
+        # there might be a registry entry pointing to the installation path. The key will
+        # differ depending on version and architecture. It seems best to let the PATH
+        # determine the file path to use since that will provide the same pwsh.exe one
+        # would invoke from any shell.
+        interpreter == "powershell" ? Chef::Util::PathHelper.join(basepath, "WindowsPowerShell", "v1.0", "#{interpreter}.exe") : interpreter
```
Comment (phiggins) [inline lib/chef/provider/powershell_script.rb:64]: This is kind of a lot for a one-liner.

**Review by phiggins — APPROVED** [review]

File: lib/chef/provider/powershell_script.rb (hunk 2 above)
Comment (mwrock) [reply lib/chef/provider/powershell_script.rb:64]: haha. I'll make it multi line. I'm used to the rust linter that would have complained if it should be broken up.

**Review by phiggins — APPROVED** [review]
Seems good to me 👍

**Pull request merged**